src/python/pants/core/target_types.py

```diff
@@ -206,5 +216,142 @@ class GenericTarget(Target):
     core_fields = (*COMMON_TARGET_FIELDS, Dependencies)
 
 
+# -----------------------------------------------------------------------------------------------
+# `archive` target
+# -----------------------------------------------------------------------------------------------
+
+# TODO(#10888): Teach project introspection goals that this is a special type of the `Dependencies`
+#  field.
+class ArchivePackages(StringSequenceField):
+    """Addresses to any targets that can be built with `./pants package`.
+
+    Pants will build the assets as if you had run `./pants package`. It will include the
+    results in your archive using the same name they would normally have, but without the `dist/`
+    prefix.
+
+    You can include anything that can be built by `./pants package`, e.g. a `python_binary`,
+    `python_awslambda`, or even another `archive`.
+    """
+
+    alias = "packages"
+
+
+# TODO(#10888): Teach project introspection goals that this is a special type of the `Dependencies`
+#  field.
+class ArchiveFiles(StringSequenceField):
```

Comment:
Ideally these two fields would be merged. Their "output type" is the same (loose files), and it's only their input field set types that might be different.

Reply:
I disagree. From a teaching perspective, I think having them be separate makes things clearer. We can have specific docstring for each type of input, rather than trying to say "You can do X _or_ Y" in the same help message, and putting all the nuance in one single message.

I also think that it's useful to make very explicit what type of dependencies we expect you to put in these fields. Let's say we merged into something like a field named `dependencies` - it would be less clear that this should only contain `files`, `relocated_files`, or types that can be built via `package`.